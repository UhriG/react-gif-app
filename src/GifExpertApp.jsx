import { useState } from 'react';
import { GifGrid, AddCategory } from './components';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState([]);

	const onAddCategory = (newCategory) => {
		// Check if the category already exists.
		if (
			categories.some(
				(category) =>
					category.toLowerCase() === newCategory.toLowerCase()
			)
		)
			return;

		// Add the new category.
		setCategories([newCategory, ...categories]);
	};

	return (
		<>
			<h1>GifExpertApp</h1>

			<AddCategory onNewCategory={onAddCategory} />

			{categories.map((category) => (
				<GifGrid key={category} category={category} />
			))}
		</>
	);
};
