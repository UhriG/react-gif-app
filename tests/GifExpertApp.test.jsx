import { screen, render } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
	test('debe de mostrarse correctamente', () => {
		const { container } = render(<GifExpertApp />);
		expect(container).toMatchSnapshot();
	});

	test('debe de mostrar una lista de categorÃ­as', () => {
		const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
		const { container } = render(
			<GifExpertApp defaultCategories={categories} />
		);

		expect(container).toMatchSnapshot();
		expect(screen.getAllByRole('img').length).toBe(categories.length);
	});
});
