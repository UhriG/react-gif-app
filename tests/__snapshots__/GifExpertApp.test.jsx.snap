// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de categor√≠as 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gif"
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`Pruebas en <GifExpertApp /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gif"
      type="text"
      value=""
    />
  </form>
</div>
`;
